src/network/postgres/postgres_network_commands.cpp

```diff
@@ -2,36 +2,80 @@
 
 #include <memory>
 #include <string>
+#include <variant>
 
+#include "network/postgres/postgres_packet_util.h"
 #include "network/postgres/postgres_protocol_interpreter.h"
-#include "parser/postgresparser.h"
+#include "network/postgres/statement.h"
 #include "traffic_cop/traffic_cop.h"
 #include "traffic_cop/traffic_cop_util.h"
 
 namespace terrier::network {
 
-/**
- * SimpleQuery always
- * @param out
- * @param connection
- * @return
- */
 static Transition FinishSimpleQueryCommand(const common::ManagedPointer<PostgresPacketWriter> out,
                                            const common::ManagedPointer<ConnectionContext> connection) {
   out->WriteReadyForQuery(connection->TransactionState());
   return Transition::PROCEED;
 }
 
-Transition SimpleQueryCommand::Exec(common::ManagedPointer<ProtocolInterpreter> interpreter,
-                                    common::ManagedPointer<PostgresPacketWriter> out,
-                                    common::ManagedPointer<trafficcop::TrafficCop> t_cop,
-                                    common::ManagedPointer<ConnectionContext> connection) {
-  const std::string query = in_.ReadString();
-  NETWORK_LOG_TRACE("Execute SimpleQuery: {0}", query.c_str());
+static void ExecutePortal(const common::ManagedPointer<network::ConnectionContext> connection_ctx,
+                          const common::ManagedPointer<Portal> portal,
+                          const common::ManagedPointer<network::PostgresPacketWriter> out,
+                          const common::ManagedPointer<trafficcop::TrafficCop> t_cop, const bool explicit_txn_block) {
+  trafficcop::TrafficCopResult result;
 
-  const auto parse_result = t_cop->ParseQuery(query, connection, out);
+  const auto query_type = portal->GetStatement()->GetQueryType();
+  const auto physical_plan = portal->PhysicalPlan();
 
-  if (parse_result == nullptr) {
+  // This logic relies on ordering of values in the enum's definition and is documented there as well.
+  if (query_type <= network::QueryType::QUERY_DELETE) {
```

Comment:
Nit: It'd be nice if this was turned into a function or a macro that made it clearer what this comparison represents like make a helper method like IsTransactionStatement(query_type) or something.